contracts/src/periphery/cert/router/EigenDACertVerifierRouter.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifier} from "src/periphery/cert/interfaces/IEigenDACertVerifier.sol";
+import {IEigenDACertVerifierRouter} from "src/periphery/cert/interfaces/IEigenDACertVerifierRouter.sol";
+import {OwnableUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+contract EigenDACertVerifierRouter is IEigenDACertVerifierRouter, OwnableUpgradeable {
+    mapping(uint32 => address) public certVerifiers;
+    uint32[] public certVerifierABNs;
+
+    event CertVerifierAdded(uint32 indexed abn, address indexed certVerifier);
+
+    error ABNNotInFuture(uint32 abn);
+    error ABNNotGreaterThanLast(uint32 abn);
+    error InvalidCertLength();
+    error NoCertVerifierAvailable();
+    error NoCertVerifierFound(uint32 rbn);
+
+    /// IEigenDACertVerifierRouter ///
+
+    function checkDACert(bytes calldata certBytes) external view returns (uint8) {
+        return IEigenDACertVerifier(getCertVerifierAt(_getRBN(certBytes))).checkDACert(certBytes);
+    }
+
+    function getCertVerifierAt(uint32 abn) public view returns (address) {
+        return certVerifiers[_findClosestRegisteredABN(abn)];
+    }
+
+    /// ADMIN ///
+
+    function initialize(address _initialOwner) external initializer {
+        _transferOwnership(_initialOwner);
+    }
+
+    function addCertVerifier(uint32 abn, address certVerifier) external onlyOwner {
+        if (abn <= block.number) {
+            revert ABNNotInFuture(abn);
+        }
```

Comment:
what's wrong with setting it at the current block number?

Reply:
Good point, and you are making me question whether this check is needed at all. Maybe the only check we need is that ABNS must be strictly increasing?

Reply:
No we definitely don't want certVerifiers to be deployed at blocks in the past! As that can change the truth of a cert that has already been verified. I think current block should be ok, but haven't thought through it 100%. @bxue-l2 have you by any chance?

Reply:
Does "current block number" refer to a block which has already been created, or the block in which this call is executing?

The `abn` must be >= to the block number in which this call is executed, but I don't understand the precise definition of `block.number`

Reply:
yes, "current block number" is the block in the call being executed. I'll make the change to a less strict equality.